### PR TITLE
Refactor validation to share same validator

### PR DESCRIPTION
**Pull request by Flanker32**
Refactor validation to share same validator

image

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/SchemaValidator.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                .forEach(pair -> registerSchema(FilenameUtils.getBaseName(pair.getKey()), pair.getValue()));
+    }
+
+    public static SchemaValidator getInstance() {
+        return LazyHolder.INSTANCE;
+    }
+
+    public void registerSchema(@Nonnull final String schemaId, @Nonnull final JsonNode schema) {
+        if (schemaMap.containsKey(schemaId)) {
+            AzureMessager.getMessager().info(String.format("Updating schema for %s", schemaId));
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/SchemaValidator.java:55]: new style format

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/SchemaValidator.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    public Set<ValidationMessage> validate(@Nonnull final String schemaId, @Nonnull final Object value) {
+        return validate(schemaId, value, "$");
+    }
+
+    public Set<ValidationMessage> validate(@Nonnull final String schemaId, @Nonnull final Object value, @Nullable final String pathPrefix) {
+        return validate(schemaId, objectMapper.convertValue(value, JsonNode.class), pathPrefix);
+    }
+
+    public Set<ValidationMessage> validate(@Nonnull final String schemaId, @Nonnull final JsonNode value) {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/SchemaValidator.java:77]: Set -> List

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/ValidationMessage.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    private Map<String, Object> details;
+    private String rawMessage;
+
+    static ValidationMessage fromRawMessage(com.networknt.schema.ValidationMessage validationMessage) {
+        return ValidationMessage.builder()
+                .type(validationMessage.getType())
+                .code(validationMessage.getCode())
+                .path(validationMessage.getPath())
+                .arguments(validationMessage.getArguments())
+                .details(validationMessage.getDetails())
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/ValidationMessage.java:28]: delete detail

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/ValidationMessage.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                .rawMessage(validationMessage.getMessage()).build();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/ValidationMessage.java:29]: delete

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/SchemaValidator.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import static com.fasterxml.jackson.databind.MapperFeature.AUTO_DETECT_CREATORS;
+import static com.fasterxml.jackson.databind.MapperFeature.AUTO_DETECT_GETTERS;
+import static com.fasterxml.jackson.databind.MapperFeature.AUTO_DETECT_IS_GETTERS;
+
+public class SchemaValidator {
+
+    private final Map<String, JsonSchema> schemaMap = new HashMap<>();
+    private final JsonSchemaFactory factory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
+    private final ObjectMapper objectMapper = new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
+            .disable(AUTO_DETECT_CREATORS, AUTO_DETECT_GETTERS, AUTO_DETECT_IS_GETTERS);
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/validator/SchemaValidator.java:42]: can all these fields be `static`

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**